packages/flutter/lib/src/cupertino/page_scaffold.dart

```diff
@@ -59,22 +69,28 @@ class CupertinoPageScaffold extends StatelessWidget {
 
       // TODO(xster): Use real size after partial layout instead of preferred size.
       // https://github.com/flutter/flutter/issues/12912
-      final double topPadding = navigationBar.preferredSize.height
-          + existingMediaQuery.padding.top;
+      final double topPadding =
+          navigationBar.preferredSize.height + existingMediaQuery.padding.top;
+
+      // TODO: refactor with MediaQuery.padding.bottom
+      final double bottomPadding = resizeToAvoidBottomPadding
+          ? existingMediaQuery.viewInsets.bottom
+          : 0.0;
 
       // If navigation bar is opaquely obstructing, directly shift the main content
       // down. If translucent, let main content draw behind navigation bar but hint the
       // obstructed area.
       if (navigationBar.fullObstruction) {
         paddedContent = new Padding(
-          padding: new EdgeInsets.only(top: topPadding),
+          padding: new EdgeInsets.only(top: topPadding, bottom: bottomPadding),
           child: child,
         );
       } else {
         paddedContent = new MediaQuery(
           data: existingMediaQuery.copyWith(
             padding: existingMediaQuery.padding.copyWith(
               top: topPadding,
+              bottom: bottomPadding,
```

Comment:
Ah my apologies, I misread your previous code. You weren't adding the MediaQuery.padding.bottom as padding, you were just propagating it down. That was correct. 

This should just be existingMediaQuery.padding.bottom + bottomPadding as you had it before.